src/main/java/de/tum/in/www1/artemis/service/TextAssessmentQueueService.java

```diff
@@ -0,0 +1,167 @@
+package de.tum.in.www1.artemis.service;
+
+import java.util.*;
+
+import org.springframework.context.annotation.Lazy;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.enumeration.Language;
+import de.tum.in.www1.artemis.repository.TextClusterRepository;
+
+@Service
+@Profile("automaticText")
+public class TextAssessmentQueueService {
+
+    private final TextClusterRepository textClusterRepository;
+
+    private final TextSubmissionService textSubmissionService;
+
+    public TextAssessmentQueueService(TextClusterRepository textClusterRepository, @Lazy TextSubmissionService textSubmissionService) {
+        this.textClusterRepository = textClusterRepository;
+        this.textSubmissionService = textSubmissionService;
+    }
+
+    /**
+     * Calculates the proposedTextSubmission for a given Text exercise
+     *
+     * @param textExercise the exercise for
+     * @throws IllegalArgumentException if textExercise isn't automatically assessable
+     * @return a TextSubmission with the highest information Gain if there is one
+     */
+    public Optional<TextSubmission> getProposedTextSubmission(TextExercise textExercise) {
+        return getProposedTextSubmission(textExercise, null);
+    }
+
+    /**
+     * Calculates the proposedTextSubmission for a given Text exercise
```

Comment:
What is a `proposedTextSubmission`?

Reply:
That is the TextSubmission which should be assessed next.

Reply:
This should somehow be reflected in the JavaDoc.